### PR TITLE
DDD-57 fetch orders

### PR DESCRIPTION
**Pull request by Zygis0321**

### CODE REVIEW

File: ui/src/components/OrderHistory.jsx [hunk 1]
```diff
@@ -1,117 +1,30 @@
...
+          responseJson.data[index].pickupDateTime = new Date(
+            responseJson.data[index].pickupDateTime
+          );
+        }
+        setOrders(responseJson.data);
+      } catch (e) {
+        displayError("Failed to load orders.");
+      }
+    })();
+  }, []);
```
Comment (Accir) [inline ui/src/components/OrderHistory.jsx]: displayError as dependency is needed here

File: ui/src/components/OrderHistory.jsx [hunk 2]
```diff
@@ -1,117 +1,30 @@
...
+  const { displayError } = useMessage();
+  useEffect(() => {
+    (async function fetchData() {
+      try {
+        const response = await fetch(ORDERS);
+        let responseJson = await response.json();
+        for (let index in responseJson.data) {
+          responseJson.data[index].pickupDateTime = new Date(
+            responseJson.data[index].pickupDateTime
+          );
```
Comment (Accir) [inline ui/src/components/OrderHistory.jsx:20]: Maybe if there is 0 orders, let's display somthing like "No orders to show"

Comment (Accir) [reply ui/src/components/OrderHistory.jsx:20]: The orders API is also being called non-stop

File: ui/src/components/OrderHistory.jsx (hunk 1 above)
Comment (Accir) [reply ui/src/components/OrderHistory.jsx]: it seems if I add the missing dependency, the API gets called non-stop

Comment (Zygis0321) [reply ui/src/components/OrderHistory.jsx]: added displayError ref to avoid this warning

**Pull request merged**